README.md

```diff
@@ -7,15 +7,198 @@
+[Bootstrap's Carousel component](http://getbootstrap.com/javascript/#carousel):
+
+![Bootstrap Carousel][carousel]
+
+A carousel is a perfect example on where using a store to extract out component
+state doesn't necessarily make things easier (or would simply be a massive
+overkill).
+
+Writing the essential handler functions for the component in question using
+"classical" React-style without **any** "outside" state is trivial:
+
+```js
+class Carousel extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			// We start out rendering the first slide. 0 denotes the index of the
+			// active item.
+			currentSlide: 0,
+		};
+	}
+	/**
+	 * Handler function that transitions to the next slide in the carousel.
+	 * This is the function that will be run once the user clicks the "next"
+	 * button.
+	 */
+	goNext = () => {
+		const previousSlide = this.state.currentSlide;
+		this.setState({ currentSlide: previousSlide + 1 });
+	}
+	/**
+	 * Equivalent to `goNext`. Handler function that will be invoked when clicking
+	 * the "back" button.
+	 */
+	goBack = () => {
+		const previousSlide = this.state.currentSlide;
+		this.setState({ currentSlide: previousSlide - 1 });
+	}
+	render() {
+		// Magic goes here
+	}
+}
+```
+
+In this case, using the local state of the component has a couple of advantages
+over using an external store:
+
+1. The state is **by definition** bound to the component
+
+When rendering a very long list of carousels, keeping the state stored in the
+store in sync with the _actual_ list of rendered components is hard. Let's say
+we render one carousel for each photo "collection" — which could for example
+be represented by an array for image sources — keeping the array length in sync
+with whatever data structure we would use in the store for representing the
+selected slide index is unnecessarily complex. E.g. if we add a photo collection,
+we need to add the `currentSlide` property to the store as well.
+
+Simply distinguishing between "component UI" state and global application state
+radically simplified the architecture in the above case, since component state
+can by definition not exist without a matching component (and vice versa).
+
+2. Simplified Testing
+
+Testing React component is extremely easy compared to other frameworks, such as
+Angular. Reason being that React components are by definition declarative, while
+Angular heavily relies on imperative APIs.
+
+Using stores doesn't necessarily break this abstraction, but it makes it much
+harder to properly test all the possible states that a component can be in,
+since a store might contain state that isn't directly consumed by the component
+to be tested.
+
+But more importantly, we now need to manage a store during testing. This means
+we need to make sure we **always** restore it to its previous state in an
+`beforeEach`, otherwise this can lead to hard to debug failed tests.
```

Comment:
I wouldn't focus specifically on `beforeEach`, since not all testing frameworks are built like this. Instead, I'd opt for a more generic 'before every test'.

(Same comment just below this line too)